src/Installer/core-sdk-tasks/ExtractArchiveToDirectory.cs

```diff
@@ -88,13 +88,22 @@ public override bool Execute()
                             {
                                 if (ShouldExtractItem(entry.FullName))
                                 {
-                                    if (!Directory.Exists(Path.Combine(DestinationDirectory, Path.GetDirectoryName(entry.FullName))))
+                                    var fullDestinationPath = Path.GetFullPath(Path.Combine(DestinationDirectory, entry.FullName));
+                                    if (!fullDestinationPath.StartsWith(Path.GetFullPath(DestinationDirectory), StringComparison.Ordinal))
```

Comment:
The vulnerability is not really applicable here. All extracted items are guarded by `ShouldExtractItem`, which checks the prefix of entry.